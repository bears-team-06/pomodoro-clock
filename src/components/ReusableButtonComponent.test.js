import React from 'react';
import { shallow } from 'enzyme';
import ButtonComponent from './ReusableButtonComponent';
describe('Button Component', () => {
    it('with text and is disabled true', () => {
        const button = shallow(<ButtonComponent label={'Button Text'} isDisabled={true} />);
        expect(button).toMatchSnapshot();
    });

    it('with text and is disabled false', () => {
        const button = shallow(<ButtonComponent label={'Button Text'} isDisabled={false} />);
        expect(button).toMatchSnapshot();
    });

    it('should callback on click handler', () => {
        const clickHandler = jest.fn()
        const button = shallow(<ButtonComponent label={'Button Text'} isDisabled={false} clickHandler={clickHandler}/>);
        button.simulate('click')
        expect(clickHandler).toHaveBeenCalledTimes(1)
    })
});