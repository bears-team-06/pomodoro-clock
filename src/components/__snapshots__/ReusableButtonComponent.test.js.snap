// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component with text and is disabled false 1`] = `
<button
  className="btn timer-button"
  disabled={false}
>
  Button Text
</button>
`;

exports[`Button Component with text and is disabled true 1`] = `
<button
  className="btn timer-button"
  disabled={true}
>
  Button Text
</button>
`;
